crates/red_knot_python_semantic/src/semantic_index/use_def/symbol_state.rs

```diff
@@ -287,18 +287,18 @@ impl SymbolBindings {
         }
     }
 
-    fn merge(&mut self, b: Self, visibility_constraints: &mut VisibilityConstraints) {
-        let a = std::mem::take(self);
+    fn merge(&mut self, mut b: Self, visibility_constraints: &mut VisibilityConstraints) {
+        let mut a = std::mem::take(self);
         self.live_bindings = a.live_bindings.clone();
         self.live_bindings.union(&b.live_bindings);
 
         // Invariant: These zips are well-formed since we maintain an invariant that all of our
         // fields are sets/vecs with the same length.
         let a = (a.live_bindings.iter())
-            .zip(a.constraints)
+            .zip(a.constraints.iter_mut())
```

Comment:
A comment here would probably be useful. It wasn't immediately clear why we use a `&mut` here even with your explanation. Should we do the same in the other `merge` method and for `visibility_constraints?

Reply:
I think it won't be as beneficial for `visibility_constraints`, since those are already smaller.  But I can check!  (And I will add the comment)

Reply:
Yep as expected, it was a wash doing the same thing to `visibility_constraints`.  (Those are 32-bit IDs, which are cheap to copy, and not 24-byte `BitSet`s.)